remotes/docker/config/hosts.go

```diff
@@ -232,11 +232,15 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 				rhosts[i].Authorizer = authorizer
 			}
 
-			// When TLS has been explicitly configured for the operation or host, use the
-			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the request as http.
-			// This allows preference for https when configured but also catches TLS errors early enough
-			// in the request to avoid sending the request twice or consuming the request body.
-			if host.scheme == "http" && explicitTLS {
+			// When TLS has been explicitly configured for the operation or host and
+			// the protocol from the port number is ambiguous, use the
+			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the
+			// request as http. This allows preference for https when configured but
+			// also catches TLS errors early enough in the request to avoid sending
+			// the request twice or consuming the request body.
+			if host.scheme == "http" && explicitTLS &&
+				strings.ContainsAny(host.host, ":") &&
```

Comment:
q: For understanding, How does presence of `:` in host make protocol ambigous?

Reply:
I'll update the parsing, it is more about determining whether the port is 80 or 443 since the protocol for those ports is not ambiguous. If a configuration is provided which defines insecure tls and http default on port 5000, then the protocol is ambiguous and the fallback should be used.

Reply:
Updated